share/cmake/cccl/cccl-config-version.cmake

```diff
@@ -0,0 +1,23 @@
+# TODO How should we encode this as a source of truth?
+set(CCCL_VERSION_MAJOR 2)
+set(CCCL_VERSION_MINOR 2)
+set(CCCL_VERSION_PATCH 0)
+set(CCCL_VERSION_TWEAK 0)
+
+set(CCCL_VERSION "${CCCL_VERSION_MAJOR}.${CCCL_VERSION_MINOR}.${CCCL_VERSION_PATCH}.${CCCL_VERSION_TWEAK}")
+
+set(PACKAGE_VERSION ${CCCL_VERSION})
+set(PACKAGE_VERSION_COMPATIBLE FALSE)
+set(PACKAGE_VERSION_EXACT FALSE)
+set(PACKAGE_VERSION_UNSUITABLE FALSE)
+
+# Semantic versioning:
+if(PACKAGE_VERSION VERSION_GREATER_EQUAL PACKAGE_FIND_VERSION)
+  if(CCCL_VERSION_MAJOR VERSION_EQUAL PACKAGE_FIND_VERSION_MAJOR)
+    set(PACKAGE_VERSION_COMPATIBLE TRUE)
```

Comment:
Are we okay stating that CCCL 3 is not eligible to match `find_package(CCL 2)`?

Reply:
Users can pull in all subproject separately with specific version check to make their weird usecases work.

If there's an example of this, we can test it and make sure that any special cases in the subproject logic are present.